R/FilterState.R

```diff
@@ -160,151 +128,251 @@ InteractiveFilterState <- R6::R6Class( # nolint
+        varname = private$get_varname(),
+        choices = private$get_choices(),
+        selected = private$get_selected(),
+        keep_na = private$get_keep_na(),
+        keep_inf = private$get_keep_inf(),
+        fixed = private$fixed
+      )
+      args <- append(args, private$extras)
+      args <- Filter(Negate(is.null), args)
+      do.call(filter_var, args)
     },
 
     #' @description
-    #' Get selected values from `FilterState`.
-    #'
-    #' @return class of the returned object depends of class of the `FilterState`
+    #' Returns reproducible condition call for current selection relevant
+    #' for selected variable type.
+    #' Method is using internal reactive values which makes it reactive
+    #' and must be executed in reactive or isolated context.
     #'
-    get_selected = function() {
-      private$selected()
+    get_call = function() {
+      stop("this is a virtual method")
     },
 
     #' @description
-    #' Returns the filtering state.
+    #' Shiny module server.
     #'
-    #' @return `list` containing values taken from the reactive fields:
-    #' * `selected` (`atomic`) length depends on a `FilterState` variant.
-    #' * `keep_na` (`logical(1)`) whether `NA` should be kept.
+    #' @param id (`character(1)`)\cr
+    #'   shiny module instance id
     #'
-    get_state = function() {
-      list(
-        selected = self$get_selected(),
-        keep_na = self$get_keep_na()
+    #' @return `moduleServer` function which returns reactive value
+    #'   signaling that remove button has been clicked
+    #'
+    server = function(id) {
+      moduleServer(
+        id = id,
+        function(input, output, session) {
+          private$server_summary("summary")
+          private$server_inputs("inputs")
+          observeEvent(input$enable,
+                       {
+                         if (isTRUE(input$enable)) {
+                           private$enable()
+                         } else {
+                           private$disable()
+                         }
+                       },
+                       ignoreInit = TRUE
+          )
+          reactive(input$remove) # back to parent to remove self
+        }
       )
     },
 
     #' @description
-    #' Prints this `FilterState` object.
-    #'
-    #' @param ... additional arguments to this method
+    #' Shiny module UI.
     #'
-    print = function(...) {
-      cat(shiny::isolate(self$format()), "\n")
+    #' @param id (`character(1)`)\cr
+    #'  shiny element (module instance) id;
+    #'  the UI for this class contains simple message stating that it is not supported
+    #' @param parent_id (`character(1)`) id of the FilterStates card container
+    ui = function(id, parent_id = "cards") {
+      ns <- NS(id)
+
+      theme <- getOption("teal.bs_theme")
+
+      if (is.null(theme)) {
+        private$ui_bs3(id, parent_id)
+      } else {
+        private$ui_bs45(id, parent_id)
+      }
```

Comment:
ui got simplied on the filter_panel_refactor@main - please keep the ui from refactor branch as distinction `ui_bs3` and `ui_bs45` was removed. Please ask @asbates for help if needed